x/wasm/keeper/api.go

```diff
@@ -22,25 +26,39 @@ const (
 var (
 	costHumanize            = DefaultGasCostHumanAddress * types.DefaultGasMultiplier
 	costCanonical           = DefaultGasCostCanonicalAddress * types.DefaultGasMultiplier
+	costValidate            = DefaultGasCostCanonicalAddress * types.DefaultGasMultiplier
 	costJSONDeserialization = wasmvmtypes.UFraction{
 		Numerator:   DefaultDeserializationCostPerByte * types.DefaultGasMultiplier,
 		Denominator: 1,
 	}
 )
 
-func humanAddress(canon []byte) (string, uint64, error) {
+func humanizeAddress(canon []byte) (string, uint64, error) {
 	if err := sdk.VerifyAddressFormat(canon); err != nil {
 		return "", costHumanize, err
 	}
 	return sdk.AccAddress(canon).String(), costHumanize, nil
 }
 
-func canonicalAddress(human string) ([]byte, uint64, error) {
+func canonicalizeAddress(human string) ([]byte, uint64, error) {
 	bz, err := sdk.AccAddressFromBech32(human)
 	return bz, costCanonical, err
 }
 
+func validateAddress(human string) (uint64, error) {
```

Comment:
Do we have a place to write unit tests for those functions?

Reply:
Currently not, but I can add an `api_test.go` and add tests there.

Reply:
Yeah that would be great. At least for the new addition

Reply:
I added a very basic test here.